source/Halibut/Transport/SecureListener.cs

```diff
@@ -111,9 +113,18 @@ async Task Accept()
                     }
                     catch (Exception ex)
                     {
-                        log.WriteException(EventType.Error, "Error accepting TCP client", ex);
-                        // Slow down the logs in case an exception is immediately encountered each AcceptTcpClientAsync
-                        Thread.Sleep(1000);
+                        numberOfFailedAttemptsInRow++;
+                        log.WriteException(EventType.Error, $"Error accepting TCP client", ex);
+                        // Slow down the logs in case an exception is immediately encountered after 3 failed AcceptTcpClientAsync calls
+                        if (numberOfFailedAttemptsInRow >= 3)
+                        {
+                            var millisecondsTimeout = Math.Max(0, Math.Min(numberOfFailedAttemptsInRow - 3, 100)) * 10;
+                            log.Write(
+                                EventType.Diagnostic,
+                                $"Connection has failed {numberOfFailedAttemptsInRow} times. Waiting {millisecondsTimeout}ms before retrying. For a detailed troubleshooting guide go to https://g.octopushq.com/TentacleTroubleshooting"
+                            );
+                            Thread.Sleep(millisecondsTimeout);
```

Comment:
Bit of a drive by, but would `Task.Delay()` be more appropriate here?

Reply:
It's much easier to see where this process is hanging, which was valuable in finding this bug. I'll do up a PR to make this method not async